File: Python/121. Best Time to Buy and Sell Stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        i = 0
        profit = 0
        minPrice=float('inf')
        for i in prices:
            minPrice=min(minPrice,i)
            profit=max(profit,i-minPrice)
        return profit

File: C#/121. Best Time to Buy and Sell Stock.cs
public class Solution {
    public int MaxProfit(int[] prices) {
        int minPrice=prices[0];
        int profit=0;
        for (int i = 0; i<prices.Length;i++){
            minPrice=Math.Min(minPrice,prices[i]);
            profit=Math.Max(profit,prices[i]-minPrice);
        }
        return profit;
    }
}
